Pull from developing branch to main branch

gruvix opened the pull request:
![imagen](https://github.com/gruvix/Simon-Dice/assets/72170232/cf7587ca-ee88-400a-badb-0e5fd2691137)

gruvix committed:
creasion js

gruvix committed:
preparado js funciones basicas

armado de funciones basicas, sin desarrollar

gruvix committed:
add random 1-4 gen

gruvix committed:
delay universal

gruvix committed:
resaltar cuadros por cada segundo

gruvix committed:
funcion para manejar rondas con turnbo jugador y turno computadora

gruvix committed:
bloqueo de input fuera del turno del jugador

gruvix committed:
color a los cuadros del simon

gruvix committed:
bloqueo y desbloqueo de input usuario

gruvix committed:
improve colores cuadros

gruvix committed:
resalta color cuando presionan los botones

gruvix committed:
Transicion de color al clickear

gruvix committed:
boton rojo pasado a rosa

gruvix committed:
resalta todos los cuadros al iniciar

gruvix committed:
fix $cuadro = null

gruvix committed:
remove null check

gruvix committed:
se borra la secuencia del jugador al terminar

gruvix committed:
Barras comentadoras y orden - edit readme

ordene todo para que sea mas facil ubicar cada pieza del programa

agregue un comentario sobre un problema de recursividad en el script

gruvix committed:
++orden

gruvix committed:
comparacion de secuencias ganar y perder

gruvix committed:
mistake and game over

gruvix committed:
rename variable

renamed variable justPlayed for nextTurno to reduce confussion, also mirrored all calls of the variable and changed the IF comparison

gruvix committed:
add mensaje error

gruvix committed:
add game finisher

gruvix committed:
replace delayFucntion con SetTimout

gruvix committed:
funciones de ocultar y mostrar

gruvix committed:
game over msj

gruvix committed:
removed delayFunction

gruvix committed:
fix SetTimout

los settimout se estaban llamando erroneamente

gruvix committed:
remove comentario recursividad

ya no tiene el problema de revursividad que tenía

gruvix committed:
wrong slection box

cuando clickeas un cuadro incorrecto se resalta de rojo para que notes el error

gruvix committed:
limpieza y timeout al boton inicio

gruvix committed:
going full english

gruvix committed:
simple style adjustments

gruvix committed:
renamed everything in english

gruvix committed:
now sequences are compared in a diff function

gruvix committed:
fix ai starts too fast

gruvix committed:
add speed setting

add speed setting to manipulate game speed
still doesent affect game speed

gruvix committed:
renamed tic into a variable

gruvix committed:
speed settings working

with options
lento (double time for all actions)
medio (original normal speed)
rapido (halft time for all actions)

gruvix committed:
highlight correct color box when clicking a wrong one

gruvix committed:
updates turn indicator

gruvix committed:
add style to indicators

gruvix committed:
now turn name has own color

gruvix committed:
fixed wrong timing for counter

gruvix committed:
Cheat system

gruvix committed:
made cheat code cleaner

gruvix committed:
remove old comments

gruvix committed:
renamed variables for easier comprehension

gruvix committed:
comments edits

gruvix committed:
cheat code next click highlight

adds a border to next box that must be clicked  for fast tracking

gruvix committed:
cheat border disappears after finishing turn

gruvix committed:
add sound files

add sounds for each color and an error sound

gruvix committed:
sounds for clicks

gruvix committed:
sounds for computer turn

gruvix committed:
bootstrap

add bootstrap to the styles, adapted html to bootstrap

gruvix committed:
improved main rectangles sizing

gruvix committed:
add style boxes for game elements

gruvix committed:
add color background to cheats

gruvix committed:
Create package.json

gruvix committed:
ignore node modules

gruvix committed:
installed bootstrap with npm

gruvix committed:
now uses local bootstrap version

gruvix committed:
now includes install instructions

fsodano commented on css/main.css:
```diff
@@ -1,6 +1,51 @@
+    border: 5px solid green;
+}
+
+#turn{
```
ojo con el spanglish, tenés cuadro y turn, player, etc.

fsodano commented on index.html:
```diff
@@ -5,30 +5,86 @@
+            Es el turno de&nbsp;<a id="turn"> - </a>
+         </div>
+         <div class="col-sm-4 alert alert-secondary" role="alert">
+            &nbsp;Ronda N#<a id="ronda">-</a>
```
esto por qué es un anchor si no es clickeable? Hacelo un <span> en todo caso

gruvix replied:
Html es lo que menos manejo
no se que sería un "un" (o quizá solo falta especificar) 
me revisé la lista entera de elementos de html y no encontré uno que sirviera específicamente para esto, ¿hay algún elemento que se use para estas situaciones que no aplique ningún estilo?

fsodano commented on index.html:
```diff
@@ -5,30 +5,86 @@
+         <div class="col-sm"></div>
+
+         <div class="col-sm">
+            <!-- Third square -->
```
suman estos comentarios? se llama id="cuadro3", no podés ser mucho más explícito que eso

fsodano commented on js/main.js:
```diff
@@ -0,0 +1,261 @@
+document.querySelector("#start").addEventListener("click", function(){
+    tic = document.querySelector("#speed").value;
```
mmm esta variable global no se si me gusta, mejor preguntarla en la función que la use

fsodano commented on js/main.js:
```diff
@@ -0,0 +1,261 @@
+    cheatHandler();
+});
+
+let tic = 1000; //controls general game speed, in miliseconds
```
entonces ponele `const gameSpeedInMs` y ahorrate el comentario

fsodano commented on js/main.js:
```diff
@@ -0,0 +1,261 @@
+    computerSequence = [];
+    playerSequence = [];
+}
+/////////////////////////////////////////////////////ROUND HANDLER////////////////////////////////////////
```
si sentís que debés dividir el archivo en secciones entonces ponelo en otro archivo y ahorrate el comentario

fsodano commented on js/main.js:
```diff
@@ -0,0 +1,261 @@
+    
+    computerSequence.push(randomColorBox());
+    computerSequence.forEach((colorBox, index) => {
+        const time = (index+0.5) * tic;
```
0.5? magic number

fsodano commented on js/main.js:
```diff
@@ -0,0 +1,261 @@
+    computerSequence.forEach((colorBox, index) => {
+        const time = (index+0.5) * tic;
+        setTimeout(highLightColor, time, colorBox);
+        setTimeout(handleSound, time+(0.5*tic), colorBox);
```
este 0.5 asumo que está relacionado al anterior, eso tiene que ir en otro lado con un nombre explícito

gruvix replied:
era para compensar una diferencia de tiempo entre la animación y el sonido pero terminé haciendo que tuviera otros problemas en velocidad lenta

fsodano commented on js/main.js:
```diff
@@ -0,0 +1,261 @@
+}
+
+function randomColorBox(){
+    return `cuadro${Math.floor(Math.random() * 4) + 1}`;
```
el 4 qué es? Y si mañana quiero 5 colores qué pasa?

fsodano commented on js/main.js:
```diff
@@ -0,0 +1,261 @@
+    return `cuadro${Math.floor(Math.random() * 4) + 1}`;
+}
+
+function reduceCounterBy100Ms(ultimoValor){
```
creo que nunca llamás a esta función

gruvix replied:
es llamada desde la funcion startSequence

gruvix committed:
renamed cuadro to rectangle to reduce spanglish

gruvix committed:
changed html elements

html file was using wrong elements
changed some clickables that were not clickable
a -> un

gruvix committed:
renamed html element to english

gruvix committed:
random color generator refactor

now randomColorBox will give a random number between the amount of current color rectangles instead of having a hardcoded number

gruvix committed:
Moved computer turn to new file

now all computer turn's related script is in its own file

gruvix committed:
add computerTurn.js to html

gruvix committed:
moved player turn to new file

moved all player turn-related script to a new file

gruvix committed:
moved cheat to new file

moved cheat's script to a new file

gruvix committed:
remove comments

removed unnecessary comments

gruvix committed:
renamed game speed variable
